cpp/lib/connection/driver.cpp

```diff
@@ -41,7 +42,7 @@ _native::Credential* Credential::getNative() const {
 
 
 Driver Driver::coreDriver(const std::string& coreAddress) {
-    auto p = _native::connection_open_core(coreAddress.c_str());
+    auto p = _native::connection_open_core_with_id(coreAddress.c_str(), "C++", TYPEDB_DRIVER_VERSION);
```

Comment:
We should variabilise the driver name as `C++` here so in the future they're easy to find/rename if we want